apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/asyncprofiler/JfrParser.java

```diff
@@ -363,16 +339,17 @@ private StackFrame resolveStackFrame(int classId, int methodName) {
         }
     }
 
-    private StringBuilder readUtf8String() {
+    private StringBuilder readUtf8String() throws IOException {
         return readUtf8String(false);
     }
 
-    private StringBuilder readUtf8String(boolean replaceSlashWithDot) {
-        int size = buffer.getShort();
+    private StringBuilder readUtf8String(boolean replaceSlashWithDot) throws IOException {
+        int size = bufferedFile.getShort();
```

Comment:
I'd add `& 0xff` to convert this to an unsigned short.
There hardly be a string larger than 32K, but just in case.

The same in `skipString`.